src/main/java/org/openrewrite/launchdarkly/RemoveBoolVariation.java

```diff
@@ -77,20 +78,21 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
             @Override
             public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                 J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
-                boolean isFirstArgumentFeatureKey =
-                        CursorUtil
-                                .findCursorForTree(getCursor(), mi.getArguments().get(0))
-                                .bind(c -> ConstantFold.findConstantLiteralValue(c, String.class))
-                                .map(featureKey::equals)
-                                .orSome(false);
-                if (methodMatcher.matches(mi) && isFirstArgumentFeatureKey) {
+                if (methodMatcher.matches(mi) && isFeatureKey(mi.getArguments().get(0))) {
```

Comment:
@shanman190 I believe this now does what you proposed in your earlier https://github.com/openrewrite/rewrite-launchdarkly/issues/23#issuecomment-2159344417 : only call `CursorUtil.findCursorForTree` once we've established that we're looking at the right method invocation, as opposed to an early call before.